---
title: "FBL-affected transcripts from Erales Riboseq"
author: "Marina Alexander"
date: "27/03/2020"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(rhdf5)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

List the contents of the .h5 file downloaded from ARChS4 webserver

```{r list the contents}
h5ls("data/human_transcript_v7.h5")

```
Retrieve Gene Expression Omnibus (GEO) Aceession from matrix
```{r GEO accession}
samples = h5read("data/human_transcript_v7.h5", "meta/Sample_geo_accession")

```
Retreive sample information and tissue from matrix
```{r sample information}
tissue = h5read("data/human_transcript_v7.h5", "meta/Sample_source_name_ch1")

```

Join Geo acession and sample info to allow searching for interesting datasets
```{r joining GEO acession and sample info}
metadata <- samples %>%
  as.tibble() %>% 
  bind_cols(as.tibble(tissue)) %>% 
  rename(GEO_accession = value) %>% 
  rename(Tissue = value1) %>%
  filter(str_detect(GEO_accession, "GSM2640632"))

 # to add in to code chunk above
          # Tissue == "lymph node"|
          # Tissue == "adipose"|
          # Tissue == "testes"|
          # Tissue == "lung" |
          # Tissue == "HeLa Cells"|
          # Tissue == "Hela cells"|
          # Tissue == "HeLa cells"|
          # Tissue == "adipose_postLPS"|
          # Tissue == "adipose tissue"|
          # Tissue == "HeLa cell" |
          # Tissue == "adipose, pre-LPS"|
          # Tissue == "kidney"|
          # Tissue == "cultured B-cells"|
          # Tissue == "breast"|
          # Tissue == "placenta"|
          # Tissue == "Macrophages"|
          # Tissue == "skeletal muscle"|
          # Tissue == "CD4+ T cells"|
          # Tissue == "T cells"|
          # Tissue == "Liver"|
          # Tissue == "NK cells"|
          # Tissue == "Dentritic cells"|
          # Tissue == "Neutrophils"|
          # Tissue == "Plasma cell"|
          # Tissue == "Thymus")

metadata
```
```{r retreive transcript information}

transcripts = h5read("data/human_transcript_v7.h5", "meta/transcripts")

```

```{r Erales RNA seq FBL knockdown}

	
GSE105248_meta<- read.csv("data/GSE105248_EralesRNAseq_metadata.csv") %>%
  select(GEO_Accession..exp., antibiotic_treatment) %>%
  rename(GEO = GEO_Accession..exp.)


sampGSE105248 = GSE105248_meta$GEO

sample_retrieve_GSE105248 = which(samples %in% sampGSE105248)

expression_GSE105248= h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(transcripts), sample_retrieve_GSE105248))
H5close()
rownames(expression_GSE105248) = transcripts
colnames(expression_GSE105248) = samples[sample_retrieve_GSE105248]

GSE105248 <- data.frame(names = row.names(expression_GSE105248), expression_GSE105248) %>% 
  filter(names == "ENST00000221801.7"| 
           names == "ENST00000442744.6"| 
           names == "ENST00000391857.8" | 
           names == "ENST00000610292.4"| 
           names == "ENST00000361445.8"|
           names == "ENST00000616727.4") %>% 
  gather(GEO, expn, -names) %>%
  mutate(transcript = str_replace_all(names, 
                                      c("ENST00000221801.7" = "FBL",
                                        "ENST00000442744.6" = "RPL40",
                                        "ENST00000391857.8" = "RPL13A",
                                        "ENST00000610292.4" = "P53",
                                        "ENST00000361445.8" = "mTOR",
                                        "ENST00000616727.4" = "MUC13"))) %>% 
  full_join(GSE105248_meta, by = "GEO") %>% 
  filter( transcript == "FBL")

plot_FBL <- ggplot(GSE105248, aes( x= transcript, y = expn, color = antibiotic_treatment)) +
  geom_boxplot() +
  labs(title = "transcripts expressed in FBL-knockdown cells", 
       x = "transcript",
       y = "expression")

ggsave(filename = "results/Erales_FBL_GSE105248.png", plot = plot_FBL, width = 15, height = 8, dpi = 600, units = "cm")


```

```{r Erales significant transcripts}

	
GSE105248_meta<- read.csv("data/GSE105248_EralesRNAseq_metadata.csv") %>%
  select(GEO_Accession..exp., antibiotic_treatment) %>%
  rename(GEO = GEO_Accession..exp.)


sampGSE105248 = GSE105248_meta$GEO

sample_retrieve_GSE105248 = which(samples %in% sampGSE105248)

expression_GSE105248= h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(transcripts), sample_retrieve_GSE105248))
H5close()
rownames(expression_GSE105248) = transcripts
colnames(expression_GSE105248) = samples[sample_retrieve_GSE105248]

GSE105248 <- data.frame(names = row.names(expression_GSE105248), expression_GSE105248) %>% 
  filter(names == "ENST00000382438.5"|
           names == "ENST00000382333.1"|
           names == "ENST00000239316.4" |
           names == "ENST00000512729.5"|
           names == "ENST00000339336.8"|
           names == "ENST00000616727.4"|
           names == "ENST00000621141.4"|
           names == "ENST00000325110.10") %>% 
  gather(GEO, expn, -names) %>%
  mutate(
    transcript = str_replace_all(names, c(
             "ENST00000382438.5" = "NKX3-2",
             "ENST00000382333.1" = "FGFBP1",
             "ENST00000239316.4" = "INSL4",
             "ENST00000512729.5" = "TRIML2",
             "ENST00000339336.8" = "AMTN",
             "ENST00000616727.4" = "MUC13 ",
             "ENST00000621141.4" = "GPR1_210",
             "ENST00000325110.10" = "HMGCS1_201"))) %>% 
  full_join(GSE105248_meta, by = "GEO") 

write_csv(GSE105248, path = "Erales_Riboseq_signf_levels.csv")

plot_GSE105248 <- ggplot(GSE105248, aes(x= transcript, y = expn, color = antibiotic_treatment)) +
  geom_boxplot() +
  labs(title = "transcripts affected in FBL-knockdown cells", 
       x = "transcript",
       y = "expression")

ggsave(filename = "results/Erales_FBL_GSE105248.png", plot = plot_FBL, width = 15, height = 8, dpi = 600, units = "cm")


```